json-modules/json/src/test/java/com/baeldung/jsonjava/ParseJsonBooleanUnitTest.java

```diff
@@ -0,0 +1,62 @@
+package com.baeldung.jsonjava;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import org.json.JSONObject;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+public class ParseJsonBooleanUnitTest {
+
+
+    @Test
+    void givenJSONString_whenParsed_correctBooleanValueReturned() {
+        String jsonString = "{\"name\":\"lorem ipsum\",\"active\":true,\"id\":1}";
+        JSONObject jsonObject = new JSONObject(jsonString);
+        boolean active = jsonObject.getBoolean("active");
+        assertTrue(active);
+    }
+
+    @Test
+    void givenJSONWithBooleanAs0Or1_whenParsed_correctBooleanValueReturned() {
+        String jsonString = "{\"name\":\"lorem ipsum\",\"active\":1,\"id\":1}";
+        JSONObject jsonObject = new JSONObject(jsonString);
+        int activeInt = jsonObject.getInt("active");
+        boolean isActive = (activeInt == 1);
+        assertTrue(isActive);
+    }
+
+    @Test
+    void givenJSONWithMixedRepresentationForBoolean_whenParsed_correctBooleanValueReturned() {
+        String jsonString = "{\"name\":\"lorem ipsum\",\"active\": 0,\"id\":1}";
+        JSONObject jsonObject = new JSONObject(jsonString);
+        Object activeObject = jsonObject.get("active");
+        boolean active;
+        if (activeObject instanceof Integer) {
```

Comment:
Please use the `instanceof` for casting instead:

```java
if (activeObject instanceof Integer value) {
    assertFalse(value == 1);
} else if (activeObject instanceof Boolean value) {
    assertFalse(value);
}
```

Also, there appears to be no need for the final assert if done this way.